_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,416 @@
+In answer to this question, running commands from within a broker is not good practice.
+It will start another JVM inside the container designated for the broker and can cause disruption, cause the container to run out of memory and so on.
+So it is always better to avoid running the command from a broker pod.
+
+So now it's time to get our interactive pod up and running. You can use the following command:
+
+```sh
+kubectl run --restart=Never --image=quay.io/strimzi/kafka:0.27.0-kafka-3.0.0 <INTERACTIVE-POD-NAME> -- /bin/sh -c "sleep 3600"
+```
+
+Wait till the pod gets into the `Ready` state. Once the pod gets into `Ready` state, now our next step will be to generate our `topics.json` file. 
+As we discussed above, this file will have the topics that we need to reassign.
+
+
+Now arises a good question. What topics require reassignment?
+So the answer to this is the topics that have their partitions assigned to broker `<CLUSTER-NAME>-kafka-3` and `<CLUSTER-NAME>-kafka-4` needs reassignment, and these topics partition should move to the remaining 3 brokers nodes in our cluster.
+
+To check the partitions details of a certain topic, we can use the `kafka-topics.sh` tool. We can run the following command from inside interactive pod after starting a shell process using `kubectl exec -ti <INTERACTIVE-POD-NAME> /bin/bash`
+:
+
+```sh
+bin/kafka-topics.sh --describe --topic my-topic-two --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092
+```
+which will give us the following output:
+
+```shell
+Topic: my-topic-two     TopicId: _aUFY9oMSjqBjvP9Ed3JDg PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
+        Topic: my-topic-two     Partition: 0    Leader: 0       Replicas: 0,3,1 Isr: 0,3,1
+        Topic: my-topic-two     Partition: 1    Leader: 1       Replicas: 1,0,4 Isr: 1,0,4
+        Topic: my-topic-two     Partition: 2    Leader: 4       Replicas: 4,1,2 Isr: 4,1,2
+        Topic: my-topic-two     Partition: 3    Leader: 2       Replicas: 2,4,3 Isr: 2,4,3
+        Topic: my-topic-two     Partition: 4    Leader: 3       Replicas: 3,2,0 Isr: 3,2,0
+        Topic: my-topic-two     Partition: 5    Leader: 0       Replicas: 0,1,4 Isr: 0,1,4
+        Topic: my-topic-two     Partition: 6    Leader: 1       Replicas: 1,4,2 Isr: 1,4,2
+        Topic: my-topic-two     Partition: 7    Leader: 4       Replicas: 4,2,3 Isr: 4,2,3
+        Topic: my-topic-two     Partition: 8    Leader: 2       Replicas: 2,3,0 Isr: 2,3,0
+        Topic: my-topic-two     Partition: 9    Leader: 3       Replicas: 3,0,1 Isr: 3,0,1
+```
+In the same way you can get the details for the other topic`my-topic` also.
+
+```sh
+Topic: my-topic TopicId: WyFKVZzLS8i54IGgm1ifrQ PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
```

Comment:
As a user, if I followed the steps above, will I also get 2 topics with only one of them spread onto all 5 brokers?

Reply:
No. I guess I ran the partition reassign command earlier that why the other topic is spread on only 3 brokers

Reply:
Since we have a full section on setting up an environment, it's a shame readers can't reproduce it and follow along.

Reply:
So should I get some fresh example?

Reply:
@mimaison Can I write a note here stating that the topic details might differ for them?